app/uk/gov/hmrc/cipphonenumberverification/services/VerifyHelper.scala

```diff
@@ -30,12 +31,20 @@ import scala.concurrent.Future
 
 abstract class VerifyHelper @Inject()(passcodeService: PasscodeService, govUkConnector: GovUkConnector) extends Logging {
 
-  protected def processResponse(res: HttpResponse,  phoneNumber: PhoneNumber)(implicit hc: HeaderCarrier) = res match {
+  protected def processResponse(res: HttpResponse, phoneNumber: PhoneNumber)(implicit hc: HeaderCarrier): Future[Result] = res match {
     case _ if is2xx(res.status) => processValidPhoneNumber(res, phoneNumber)
     case _ if is4xx(res.status) => Future(BadRequest(res.json))
   }
 
-  private def processValidPhoneNumber(res: HttpResponse, phoneNumber: PhoneNumber)(implicit hc: HeaderCarrier) = (res.json \ "phoneNumberType").as[String] match {
+  protected def processPasscode(enteredPasscode: Passcode, storedPasscodeOpt: Option[Passcode]): Future[Result] = storedPasscodeOpt match {
+    case Some(storedPasscode) if enteredPasscode.otp.equals(storedPasscode.otp) => passcodeService.deletePasscode(enteredPasscode) map {
```

Comment:
please can we put the does the passcode match logic in a separate method.

Reply:
is it a passcode or a otp?  the code seems inconsistent.  also the passcode object has a phone number and a passcode, so its more than a passcode, therefore the naming seems incorrect.

Reply:
I didn't change the names in this class, I just moved the protected members to the top. But happy to change

Reply:
yeah I know its not to do with this pull request as such, just a thought that its a bit misleading as its an object that has a phone number and passcode combination.